src/huggingface_hub/keras_mixin.py

```diff
@@ -16,6 +20,129 @@
 
 logger = logging.getLogger(__name__)
 
+if is_tf_available():
+    import tensorflow as tf
+
+
+def _extract_hyperparameters_from_keras(model):
+    hyperparameters = dict()
+    if hasattr(model, "optimizer"):
+        hyperparameters["optimizer"] = model.optimizer.get_config()
+    else:
+        hyperparameters["optimizer"] = None
+    hyperparameters[
+        "training_precision"
+    ] = tf.keras.mixed_precision.global_policy().name
+
+    return hyperparameters
+
+
+def _parse_model_history(model):
+    logs = model.history.history
+    num_epochs = len(logs["loss"])
```

Comment:
Careful here too, another assumption that caused my keras integration overview notebook to break. but this one is a bit easier to fix...my model had an empty `model.history.history` dict after training (its the same as the autoencoder example in the keras docs). Maybe I'm using a different tf version? (2.8.0)

Reply:
I noticed theres a model.history.params['epochs'], but my epochs shown here was 1, whereas I trained it for 10 epochs...weird.

Reply:
I don't know about the second one, but the first one I'll just assume I don't know if the model is fitted or not.

Reply:
I also tested on the autoencoder that's why I'm surprised! I tested on two models, one was sequential and one was functional. 

again this

> for the [notebook](https://gist.github.com/nateraw/2ffbd8baf8caecbbcfbdf78cda3a7090) I just don't get why your model returns an empty history though, I tested it on an autoencoder and a CNN (one was sequential and another one was functional) and it was working just fine. I just don't get the case (I get the built one but others I just don't)

I don't think it's about TF version. I think I'll just try to get `history.history` and pass plotting model card. So many edge cases I could hit there.

Reply:
You can't dump `model.history.history` to an object and call `model.history.history` again, in case anyone wonders why it returns empty in the notebook. 
<img width="621" alt="Ekran Resmi 2022-02-23 21 45 54" src="https://user-images.githubusercontent.com/53175384/155386575-191de2d1-40bb-4026-9db3-4fb1d35dff93.png">

in `push_to_hub_keras()` we already call it once and it's dumped to logs. but still if I can't get it I won't create card, given these edge cases.